src/routing_membrane.rs

```diff
@@ -950,25 +952,29 @@ impl<F> RoutingMembrane<F> where F: Interface {
         self.send_swarm_or_parallel_or_relay(&message)
     }
 
+    #[allow(unused_variables)]
     fn handle_put_data_response(&mut self, signed_message: SignedMessage,
                                            message: RoutingMessage,
                                            response: ErrorReturn) -> RoutingResult {
         info!("Handle PUT data response.");
+        let our_authority = our_authority(&message, &self.routing_table);
         let from_authority = message.from_authority();
         let from = message.source.clone();
 
-        let method_call = self.mut_interface()
-            .handle_put_response(from_authority, from, response.error);
-
-        match method_call {
-            MethodCall::Put { destination: x, content: y, } => self.put(x, y),
-            MethodCall::Get { name: x, data_request: y, } => self.get(x, y),
-            MethodCall::Refresh { type_tag, from_group, payload } => self.refresh(type_tag, from_group, payload),
-            MethodCall::Post { destination: x, content: y, } => self.post(x, y),
-            MethodCall::Delete { name: x, data : y } => self.delete(x, y),
-            MethodCall::None => (),
-            MethodCall::Forward { destination } =>
-                ignore(self.forward(&signed_message, &message, destination)),
+        for method_call in self.mut_interface().handle_put_response(from_authority, from, response.error.clone()) {
+            match method_call {
+                MethodCall::Put { destination: x, content: y, } => self.put(x, y),
+                MethodCall::Get { name: x, data_request: y, } => self.get(x, y),
+                MethodCall::Refresh { type_tag, from_group, payload } => self.refresh(type_tag, from_group, payload),
+                MethodCall::Post { destination: x, content: y, } => self.post(x, y),
+                MethodCall::Delete { name: x, data : y } => self.delete(x, y),
+                MethodCall::None => (),
+                MethodCall::Forward { destination } =>
+                    ignore(self.forward(&signed_message, &message, destination)),
+                MethodCall::Reply { data } =>
```

Comment:
this should probably not be supported, "put_response reply" as data is discarded